### PR TITLE
Fix update gripper20231204

### PR DESCRIPTION
**Pull request by LLipter**
When a parameter inside `gripperInfo/customParameters` is modified, openrave will wipe out other members after loading diff notification received from webstack. This PR fixes the issue by recursively updating the custom parameter.

### CODE REVIEW

**Review by yoshikikanemoto — APPROVED** [review]

**Comment by yoshikikanemoto** [comment]
LGTM. @rdiankov can you check and merge?

File: src/libopenrave/robot.cpp [hunk 1]
```diff
@@ -77,7 +89,10 @@ void RobotBase::GripperInfo::DeserializeJSON(const rapidjson::Value& value, dRea
         if (memberName == "id" || memberName == "name" || memberName == "grippertype" || memberName == "gripperJointNames") {
             continue;
         }
-        orjson::SetJsonValueByKey(docGripperInfo, memberName, it->value);
+
+        // update objects recursively 
+        _PartialUpdate(docGripperInfo, memberName.c_str(), it->value, docGripperInfo.GetAllocator());
```
Comment (ziyan) [inline src/libopenrave/robot.cpp:94]: @LLipter We already have `OpenRAVE::orjson::UpdateJson` why don't you use it instead?

Comment (ziyan) [reply src/libopenrave/robot.cpp:94]: Or, add the recursive version of it to `openravejson.h` instead

Comment (LLipter) [reply src/libopenrave/robot.cpp]: I moved the function to `openravejson.h` as you suggested. `OpenRAVE::orjson::UpdateJson` is not enough as it only updates the top-level key-value pair. We need to update all levels recursively.

**Comment by yoshikikanemoto** [comment]
@Puttichai this one

**Comment by yoshikikanemoto** [comment]
@ziyan can you check the latest change and approve?

**Pull request merged**